cargo-concordium/src/build.rs

```diff
@@ -798,16 +854,34 @@ pub(crate) fn build_and_run_integration_tests(
 /// The `seed` argument allows for providing the seed to instantiate a random
 /// number generator. If `None` is given, a random seed will be sampled.
 pub fn build_and_run_wasm_test(extra_args: &[String], seed: Option<u64>) -> anyhow::Result<bool> {
-    let metadata = MetadataCommand::new()
-        .exec()
-        .context("Could not access cargo metadata.")?;
+    let mut args = std::env::args().skip_while(|val| !val.starts_with("--manifest-path"));
+
+    let mut cmd = MetadataCommand::new();
+    match args.next() {
+        Some(ref p) if p == "--manifest-path" => {
+            // If a `--manifest-path` is provided use this path as the package root to
+            // test the project.
+            cmd.manifest_path(args.next().context(
+                "The argument '--manifest-path <manifest-path>' requires a value but none was \
+                 supplied.",
+            )?);
+        }
+        Some(p) => {
+            // If a `--manifest-path` is provided use this path as the package root to
+            // test the project.
+            cmd.manifest_path(p.trim_start_matches("--manifest-path="));
```

Comment:
Actually this is a bit wrong, in the sense that it will do the wrong thing if they happen to write `--manifest-pathblabh` as an argument.

It would be better to do
```rust
p.strip_prefix("--manifest-path=").context("Incorrect manifest-path flag.")?
```

Reply:
cargo run concordium test -- --manifest-pathdsaf

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/cargo-concordium concordium test -- --manifest-pathdsaf`
Error: Could not build and run tests.

Caused by:
    0: Could not access cargo metadata.
    1: `cargo metadata` exited with an error: error: unexpected argument '--manifest-pathdsaf' found
       
         note: argument '--manifest-path' exists
       
       Usage: cargo metadata --format-version <VERSION> --manifest-path <PATH>
       
       For more information, try '--help'.
```

It seems to give you some output that the flag name is wrong and should be corrected